NuRadioMC/utilities/Veff.py

```diff
@@ -1,60 +1,168 @@
 import numpy as np
 import h5py
 from scipy import interpolate
+from scipy.interpolate import interp1d
 import glob
 from six import iteritems
 import json
 import os
 import copy
+import time
 
 from NuRadioReco.utilities import units
-from NuRadioMC.EvtGen.generator import get_projected_area_cylinder, get_projected_area_cylinder_integral
 
 import logging
 logger = logging.getLogger("Veff")
 logging.basicConfig()
 
+
 # collection of utility function regarding the calculation of the effective volume of a neutrino detector
+def remove_duplicate_triggers(triggered, gids):
+    """
+    remove duplicate entried from triggered array
+    
+    The hdf5 file contains a line per shower. One event can contain many showers, i.e. if we count all triggeres
+    from all showers we overestimate the effective volume. This function modifies the triggered array such
+    that it contains not more than one True value for each event group. 
+    
+    Parameters
+    ----------
+    triggered: array of bools
+        
+    gids: array of ints
+        the event group ids
+        
+    Returns: array of floats
+        the corrected triggered array
+    """
+    gids = np.array(gids)
+    triggered = np.array(triggered)
+    uids, unique_mask, inv_mask, counts = np.unique(gids, return_index=True, return_inverse=True, return_counts=True)
+    for gid in uids[counts > 1]:
+        mask = gids == gid
+        if(np.sum(triggered[mask]) > 1):
+            idx = np.arange(len(triggered), dtype=np.int)[mask][triggered[mask] == True][1:]
+            triggered[idx] = False
+    return triggered
 
+# def get_triggered(fin):
```

Comment:
Why is this commented out and not removed? Do we anticpiate needed it later?

Reply:
this logic is not used anymore. Now the function above is used.  I just didn't delete it yet completely as this code snippet might be useful in the future. I'll remove it